src/ng-select/ng-select.component.ts

```diff
@@ -545,58 +544,57 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
         }
     }
 
-    private selectWriteValue(value: any | any[]) {
-        if (!value) {
+    private _selectWriteValue(ngModel: any | any[]) {
+        if (!ngModel) {
             return;
         }
 
         const select = (val: any) => {
             const item = this.itemsList.findItem(val, this.bindValue);
             if (item) {
                 this.itemsList.select(item);
-            } else if (val instanceof Object) {
-                const newItem = this.itemsList.addItem(val);
-                this.itemsList.select(newItem);
+            } else if (!item && val instanceof Object) {
+                this.itemsList.select({value: {...val}});
```

Comment:
this won't work since `label` property is missing. 
<img width="843" alt="screen shot 2018-01-27 at 18 12 21" src="https://user-images.githubusercontent.com/17824796/35473779-a96819dc-038d-11e8-8075-4ae376c7d9f9.png">

Ideally, you need to reuse `_mapItems` logic in `itemsList`.